### ISSUE TITLE
"Attribute name must contain word characters only" when using non-english characters in GridView

### ISSUE THREAD

**Issue by amlopezalonso** [yiisoft/yii2#8508]
MySQL allows the use of non-english characters in column names but attributes in GridView don't. For instance, using *año* (spanish for *year*) makes Yii 2.0.4 to spit an *Attribute name must contain word characters only* error.

**Comment by cebe** [comment]
whats the point in using non-ASCII characters in column names? This makes many places in the implementation very difficult. Not sure we really want to support this...

**Comment by amlopezalonso** [comment]
I agree this is not quite a critical issue but as long as MySQL supports it I just designed my schema to take advantage of it. Moreover, gii had no trouble with it and when I thought everything was about to be up and running I faced this issue which made me go back and change my db schema.

Fair enough for me If you don't want to support it but IMHO gii should  warn the developer at scaffolding stage instead of going further as if everything was ok.

**Comment by amlopezalonso** [comment]
Using non-ASCII characters may be useful in the case of some words which cannot be comfortably replaced. Most prominent example in spanish is "año" which is often replaced by "anyo", "anno" or "ano" (the latter being the spanish translation of "anus", so...).

**Comment by yurii-github** [comment]
just for fun. it is possible though.

	class test
	{
		private $props=[];
		
		function __get($n)
		{
			if (array_key_exists($n, $this->props)) {
				return $n . ' : ' . $this->props[$n] . ' : get'. '<br/>';
			}
			//error
		}
		
		function __set($n,$v)
		{
			$this->props[$n] = $v . ' : set ';
		}
	}
	
	
	$t = new test();
	$t->p1 = 'val1';
	echo $t->p1;
	
	$t->año = 'val2'; // spanish
	echo $t->año;
	
	$t->{'año año'} = 'val2'; // spanish . we can do even like this
	echo $t->{'año año'};
	
	$t->рік = 'val3'; // ukrainian
	echo $t->рік;
	
	// ukrainian vars
	$тест = new test();
	$тест->рік = '2015'; 
	echo $тест->рік;

**Comment by yurii-github** [comment]
this code

		$ar = Test::find()->select(['*'])->one();
		$f = ActiveForm::begin();
		echo $f->field($ar, 'рік')->label('Your Name');
		ActiveForm::end();

will produce

	<form id="w0" action="/mylibrary-yii2/public/" method="post"><div class="form-group field-test-���">
	<label class="control-label" for="test-���">Your Name</label>
	<input type="text" id="test-���" class="form-control" name="Test[рік]" value="2010">

	<div class="help-block"></div>
	</div></form>

**Comment by yurii-github** [comment]
using mb fixes bad chars. so, you can change it on your yii2 copy to get such functionality.
no warranty though from me :)

**Comment by amlopezalonso** [comment]
@yurii-github, thanks for pointing this out.

**Comment by terabytesoftw** [comment]
Thanks for the report. This issue is no longer relevant in current Yii2. Unicode attribute names were already fixed in #4408 and released in Yii 2.0.12.
